src/components/Api.js

```diff
@@ -0,0 +1,162 @@
+export default class Api {
+  constructor(apiSettings) {
+    this._apiSettings = apiSettings;
+    this._headers = apiSettings.headers;
+    this._baseUrl = apiSettings.baseUrl;
+    this._userInfoUrl = `${this._baseUrl}/users/me`;
+    this._cardsUrl = `${this._baseUrl}/cards`;
+    this._avatarUrl = `${this._userInfoUrl}/avatar`;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      throw response;
+    }
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(this._cardsUrl, {
+        headers: this._headers,
+      });
+      this._checkResponse(res);
+      const cardsArray = await res.json();
+      console.log(cardsArray);
+      return cardsArray;
+    } catch (err) {
+      console.log("Failed to get cards: ", err);
+      throw new Error(err.status);
+    }
+  }
+
+  //cardsArray is an array of cardata objects
+  async addArrayOfCards(cardsArray) {
+    cardsArray.forEach(async (cardData) => {
+      this.addCard(cardData);
+    });
+  }
+
+  //cardData is an object with name and link properties
+  async addCard(cardData) {
+    try {
+      const res = await fetch(this._cardsUrl, {
+        method: "POST",
+        headers: this._headers,
+        body: JSON.stringify(cardData),
+      });
+      this._checkResponse(res);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.